docs/proposals/portable-service-assets.md

```diff
@@ -0,0 +1,84 @@
+# Portable Service Assets
+
+## Introduction
+Service classes in the Service Catalog require a number of UI graphical assets (i.e. service icon, company logo, etc).  For APBs that are sourced from the RHCC, we do not want to reference these externally on a remote (non-RH managed endpoint).  Additionally, we want to be able to handle off-network / disconnected clusters thus making this a truly portable solution.
+
+## Problem Description
+Allow APBs to specify static file assets which will be accessible to any computer using OpenShift.
+* How do we build and store these assets for each APB?
+* How do we download these assets before downloading the entire APB since they will be needed before deploying?
+* How do we serve these assets?
+
+## Building and Storing Assets
+Assets will be stored under the APB folder in a folder named `assets`. As part of the `apb build` process a container image will be created copying all files from the `assets` folder into the container.
+
+```Dockerfile
+# Assets Dockerfile
+FROM scratch
+COPY assets /assets
+CMD ["noop"]
+```
+
+This docker container will be built with the same image name, but the tag will be prefixed with `_assets_`.  APB tooling will perform `docker build -t hello-world-apb:_assets_latest .`  and keep track of the image ID.
+
+After building the assets image, the tooling will also add a label to the APB image with the assets image ID.  This will ensure that if the APB is updated,  the matching assets will be associated and we can detect changes in the assets even if the APB spec has not changed.
+
+```bash
+docker build . -t docker.io/cfchase/hello-world-apb:latest --label "com.redhat.apb.assetsImageId=<ImageId>"
+```
+
+## Specifying Assets
+In a given apb.yml the user can specify `{{ assets }}` inside any metadata string values. Currently, only metadata is planned for support, but we could parse more if the use case is made.
+
+```yaml
+#apb.yml
+name: hello-world-apb
+...
+metadata:
+  imageUrl: "{{ assets }}/my-pic.svg"
+  documentationUrl: "{{ assets }}/docs.html"
+```
+
+## Downloading Assets
+After every bootstrap of apb specs, ASB will begin downloading assets in an asynchronous work job.  For each spec, ASB will check to see if there is an associated assets image ID.  If there is none, no assets are downloaded.  If there is an assets image ID and the assets are missing or the ID has changed from the last download, it will download them from the built image assets (e.g. `hello-world-apb:_assets_latest`) using commands mimicking the following:
```

Comment:
first use of ASB, change it to Ansible Service Broker (ASB), then use ASB in the rest of the document.